pkgpanda/__init__.py

```diff
@@ -355,6 +359,9 @@ def integrity_check(self):
     # If the package is already in the repository does a no-op and returns false.
     # Returns true otherwise.
     def add(self, fetcher, id, warn_added=True):
+        # Validate the package id.
+        PackageId(id)
```

Comment:
Just sending it through the constructor only to validate (even though supported by the comment) does not seem ideal.

* Better would be an introduction of a static method, that `PackageId.isvalid(id)` and use is here and in line 330.